templates/install_puppet.bat.erb

```diff
@@ -23,7 +23,7 @@ REM This may fail on agents without pxp-agent, but since this is not
 REM run interactively and the next command sets ERRORLEVEL, it's OK.
 net stop pxp-agent
 
-start /wait msiexec.exe /qn /norestart /i "<%= @_msi_location %>" /l*vx "<%= @_logfile %>" PUPPET_MASTER_SERVER="<%= @_puppet_master %>"
+start /wait msiexec.exe /qn /norestart /i "<%= @_msi_location %>" /l*vx "<%= @_logfile %>" PUPPET_MASTER_SERVER="<%= @_puppet_master %>" PUPPET_AGENT_ENVIRONMENT="<%= @environment %>"
```

Comment:
The warning issue was fixed in Puppet 4.2.2 ([PUP-3318](https://tickets.puppetlabs.com/browse/PUP-3318)).

In more recent versions, if Puppet is successfully running and `$::environment` hasn't been overridden by a custom fact, then the directory should exist. That probably doesn't hold true for Puppet 3.7 and older (not quite sure about 3.8).